clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/securitygroup/MigrateSecurityGroupAtomicOperation.groovy

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2016 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.aws.deploy.ops.securitygroup
+
+import com.netflix.spinnaker.clouddriver.aws.deploy.description.MigrateSecurityGroupDescription
+import com.netflix.spinnaker.clouddriver.aws.deploy.handlers.MigrateSecurityGroupStrategy
+import com.netflix.spinnaker.clouddriver.data.task.Task
+import com.netflix.spinnaker.clouddriver.data.task.TaskRepository
+import com.netflix.spinnaker.clouddriver.orchestration.AtomicOperation
+import org.springframework.beans.factory.annotation.Autowired
+
+import static com.netflix.spinnaker.clouddriver.aws.deploy.ops.securitygroup.SecurityGroupLookupFactory.SecurityGroupLookup
+
+class MigrateSecurityGroupAtomicOperation implements AtomicOperation<Void> {
+
+  final MigrateSecurityGroupDescription description
+
+  MigrateSecurityGroupAtomicOperation(MigrateSecurityGroupDescription description) {
+    this.description = description
+  }
+
+  @Autowired
+  SecurityGroupLookupFactory securityGroupLookupFactory
+
+  @Autowired
+  MigrateSecurityGroupStrategy migrationStrategy
```

Comment:
if this is declared as `javax.inject.Provider<MigrateSecurityGroupStrategy>` and MigrateSecurityGroupStrategy becomes a prototype scoped bean then you'd get a new one with each call to `.get()`